### PR TITLE
Grid view improvements

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
Well this ended up being a somewhat random assortment of changes... I can separate them into different PRs if needed.

* Removes the "spinner" from the grid fragment which was actually just a semi transparent black view that covered the screen :man_shrugging: 
* Changes card sizes in grids to be based on screen size (<1080p = large cards, <4k = medium cards, 4k+ = small cards) this makes much more sense than basing it on the number of items in a library imho
* Reduces chunk size in grids from 150 => 50 items. The serialization of a large number of items was taking a significant amount of time on older devices. In my testing this brought load times down from ~15s to ~5s.
* Changes the display preferences screen to use the constant values instead of hardcoded string values.

(Removed blur hash changes from this PR to be handled in #905)

**Issues**
Fixes the dimming issue in #805, but not the first issue.

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java [hunk 1]
```diff
@@ -349,14 +341,18 @@ protected int getCardHeight(String heightSetting) {
...
     }
 
-    protected int getAutoCardHeight(Integer size) {
-        Timber.d("Result size for auto card height is %d", size);
-        if (size > 35)
-            return getCardHeight(PosterSize.SMALL);
-        else if (size > 10)
-            return getCardHeight(PosterSize.MED);
-        else
+    protected int getAutoCardHeight() {
```
[the remark this answers is not in the document]
Comment (thornbill) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: Good call. I believe it should be right now.

[the remark this answers is not in the document]
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: Are you sure you can use the same values as before? AFAIK my 4k TV in dpi is around 1200dp or something like that.

[the remark this answers is not in the document]
Comment (thornbill) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: That is a good point... I really don’t have a good idea of how to know reasonable values for these considering that though. 🤔

[the remark this answers is not in the document]
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: Yeah fair point, a good solution would probably refactor a lot of the calculations in this file. To get the best sizing we should calculate based on height when using horizontal mode and width for vertical mode. Then device how many rows/columns we want, calculcate the space we have with the padding/gutter size removed and then we can determine the size/width of an individual card... right now that is definitely not how this fragment works.

[the remark this answers is not in the document]
Comment (thornbill) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: What if we just use the "medium" size as the default for now? :rofl:

[the remark this answers is not in the document]
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java:344]: Yeah that sounds good lol

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**